src/suas_planning/src/GlobalWaypointPlanner.cpp

```diff
@@ -147,57 +147,55 @@ void GlobalWaypointPlanner::ExpandObstaclesByRadius(std::vector<Obstacle>& obsta
 // not meant to be used prior to ComputeMapSize
 // This is probably a terrible way of doing this, maybe just have obstacles be copied instead
 // of being passed by reference.
-unsigned int GlobalWaypointPlanner::ComputeMapWidth(std::vector<Obstacle>& obstacles) {
-    int min_x = start_.x_;
-    int max_x = goal_.x_;
-
-    // iterate through obstacles to compute maximum size
-    std::vector<Obstacle>::iterator obstacle_it;
-    for (obstacle_it = obstacles.begin(); obstacle_it != obstacles.end(); obstacle_it++) {
-        Obstacle curr_obstacle = *obstacle_it;
-        int current_min_x = curr_obstacle.GetMinX();
-        int current_max_x = curr_obstacle.GetMaxX();
-        if (current_min_x < min_x) {
-            min_x = current_min_x;
-        }
-        if (current_max_x > max_x) {
-            max_x = current_max_x;
-        }
+unsigned int GlobalWaypointPlanner::ComputeMapWidth(
+        std::vector<Obstacle>& obstacles,
+        std::vector<Waypoint>& waypoints) {
+    std::vector<int> x_points = {start_.x_, goal_.x_};
+    // push back all y values for obstacles
+    for (Obstacle& obstacle : obstacles) {
+        x_points.push_back(obstacle.GetMinX());
+        x_points.push_back(obstacle.GetMaxX());
     }
-
-    return abs(max_x - min_x);
+    // push back all y values for waypoints
+    for (Waypoint& waypoint : waypoints_) {
+        x_points.push_back(waypoint.x_);
+        x_points.push_back(waypoint.x_);
+    }
+    // find smallest and largest x to compute range
+    const auto min_x = std::min_element(x_points.begin(), x_points.end());
+    const auto max_x = std::min_element(x_points.begin(), x_points.end());
```

Comment:
Slight typo here, probably wanted max_element

Reply:
Fixed in 3071ee2